tests/api2/test_bootenv.py

```diff
@@ -1,24 +1,70 @@
+import errno
+
+import pytest
+
+from middlewared.service_exception import ValidationErrors, ValidationError
 from middlewared.test.integration.utils import call, ssh
 
 
+def test_get_default_environment_and_make_new_one():
+    active_be_id = call('bootenv.query', [['activated', '=', True]], {'get': True})['id']
+
+    # create duplicate name to test failure
+    with pytest.raises(ValidationErrors) as ve:
+        call('bootenv.create', {'name': active_be_id, 'source': active_be_id})
+    assert ve.value.errors == [
+        ValidationError('bootenv_create.name', f'The name "{active_be_id}" already exists', errno.EEXIST)
+    ]
+
+    # create new bootenv and activate it
+    call('bootenv.create', {'name': 'bootenv01', 'source': active_be_id})
+    call('bootenv.query', [['name', '=', 'bootenv01']], {'get': True})
+    call('bootenv.activate', 'bootenv01')
+
+
+# Update tests
+def test_cloning_a_new_boot_environment():
+    call('bootenv.create', {'name': 'bootenv02', 'source': 'bootenv01'})
+    call('bootenv.activate', 'bootenv02')
+
+
+def test_change_boot_environment_name_and_attributes():
+    call('bootenv.update', 'bootenv01', {'name': 'bootenv03'})
+    call('bootenv.set_attribute', 'bootenv03', {'keep': True})
+    call('bootenv.activate', 'bootenv03')
+    assert call('bootenv.query', [['activated', '=', True]], {'get': True})['id'] == 'bootenv3'
+
+
+# Delete tests
+def test_activate_original_bootenv():
+    be_id = call('bootenv.query', [['name', '!=', 'bootenv03']], {'get': True})["id"]
+    call('bootenv.activate', be_id)
+
+
+def test_removing_boot_environments():
+    call('bootenv.set_attribute', 'bootenv03', {'keep': False})
+    job_id = call('bootenv.delete', 'bootenv02', job=True)
+    job_id = call('bootenv.delete', 'bootenv03', job=True)
```

Comment:
No longer need the `job_id` variable.

(FWIW, `flake8` would have caught this, if run locally.  IIRC doesn't run as _vigorously_ in github for tests.  Anyway, personally I wouldn't NACK the PR because of this, so your call as to whether you want to bother addressing.)

Reply:
Gotcha! I'll get those removed, good catch!